### PR TITLE
elide unused imports in ES6 emit

### LINKED ISSUE #7871: [Transforms] unused import bindings/imported namespaces are not elided
**Expected**:

```js
//// [es6ImportDefaultBindingFollowedWithNamespaceBinding_1.ts]
import defaultBinding, * as nameSpaceBinding  from "es6ImportDefaultBindingFollowedWithNamespaceBinding_0";
var x: number = defaultBinding;

//// [es6ImportDefaultBindingFollowedWithNamespaceBinding_1.js]
import defaultBinding from "es6ImportDefaultBindingFollowedWithNamespaceBinding_0";
var x = defaultBinding;
```

**Actual**:

```js
//// [es6ImportDefaultBindingFollowedWithNamespaceBinding_1.ts]
import defaultBinding, * as nameSpaceBinding  from "es6ImportDefaultBindingFollowedWithNamespaceBinding_0";
var x: number = defaultBinding;

//// [es6ImportDefaultBindingFollowedWithNamespaceBinding_1.js]
import defaultBinding, * as nameSpaceBinding from "es6ImportDefaultBindingFollowedWithNamespaceBinding_0";
var x = defaultBinding;
```

Failed tests:
*  tests\cases\compiler\es6ImportNamedImportInExportAssignment.ts

### PR DESCRIPTION
**Pull request by vladima**
fixes #7871

### CODE REVIEW

File: src/compiler/transformers/module/es6.ts [hunk 1]
```diff
@@ -19,21 +19,64 @@ namespace ts {
...
+            }
+            const newNamedBindings = visitNode(node.namedBindings, visitor, isNamedImportBindings, /*optional*/ true);
+            return newDefaultImport !== node.name || newNamedBindings !== node.namedBindings
+                ? createImportClause(newDefaultImport, newNamedBindings)
+                : node; 
+        }
+
+        function visitNamedBindings(node: NamedImportBindings): VisitResult<NamedImportBindings> {
+            if (node.kind === SyntaxKind.NamespaceImport) {
+                return resolver.isReferencedAliasDeclaration(node) ? node: undefined;
```
Comment (mhegazy) [inline src/compiler/transformers/module/es6.ts]: `return visitImportSpecifier(node);` ?

Comment (mhegazy) [inline src/compiler/transformers/module/es6.ts]: nevermind

**Comment by mhegazy** [comment]
:+1:

File: src/compiler/transformers/module/es6.ts [hunk 2]
```diff
@@ -19,21 +19,64 @@ namespace ts {
...
             return node;
         }
 
         function visitImportDeclaration(node: ImportDeclaration) {
-            if (node.importClause && !resolver.isReferencedAliasDeclaration(node.importClause, /*checkChildren*/ true)) {
-                return undefined;
+            if (node.importClause) {
+                const newImportClause = visitNode(node.importClause, visitor, isImportClause);
+                if (!newImportClause.name && !newImportClause.namedBindings) {
+                    return undefined;
```
Comment (DanielRosenwasser) [inline src/compiler/transformers/module/es6.ts]: Doesn't this mean you're eliding all `import "foo"` imports?

Comment (vladima) [inline src/compiler/transformers/module/es6.ts]: no, this is applied only for import declarations that have import clause

**Pull request merged**